ivy/functional/frontends/jax/numpy/indexing.py

```diff
@@ -80,3 +80,33 @@ def diag_indices_from(arr):
         raise ValueError("All dimensions of input must be of equal length")
     idx = ivy.arange(n, dtype=int)
     return (idx,) * ndim
+
+
+@to_ivy_arrays_and_back
+def apply_over_axes(func, a, axes):
+    for axis in axes:
+        b = func(a, axis=axis)
+        if ivy.get_num_dims(b) == ivy.get_num_dims(a):
+            a = b
+        elif ivy.get_num_dims(b) == ivy.get_num_dims(a) - 1:
+            a = ivy.expand_dims(b, axis=axis)
+        else:
+            raise ValueError("function is not returning an array of the correct shape")
+    return a
+
+
+@to_ivy_arrays_and_back
+def apply_over_axes(func, a, axes):
```

Comment:
Same function with different implementation is added above, could you please remove one of them?

Reply:
@zhumakhan done however still need some insight on getting the test to pass